docs/source/fx.rst

```diff
@@ -293,10 +293,188 @@ Writing Transformations
 
 TODO
 
-Debugging Transformations
--------------------------
+Debugging
+-----------
+
+Introduction
+^^^^^^^^^^^^^^^^
+
+Once we've generated Python code from our symbolically-traced Module,
+we'll want to verify that the proper semantics were preserved. If they
+weren't, we may need to do some debugging. The key is to work
+backwards: first, check the results of the generated module, then debug
+the generated code, then debug the process of transformations that lead
+to the generated code.
+
+If you’re not familiar with debuggers, please see the auxiliary section
+:ref:`Available Debuggers`.
+
+Debugging the Generated Code
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Because FX generates the ``forward()`` function on GraphModules, using
+traditional debugging techniques like ``print`` statements or ``pdb`` is
+ not as straightfoward. Luckily, we have several techniques we can use
+for debugging the generated code.
+
+``pdb``
+~~~~~~~~~~~~
+Use ``pdb`` to step into the running program. Although the code that
+represents the FX graph is not in any source file, we can still step
+into it manually using ``pdb`` when the forward pass is invoked.
+
+::
+    x, y = torch.rand(2, 3), torch.rand(2, 3)
+    m = symbolic_trace(M())
+    import pdb; pdb.set_trace()
+    m(x, y)
+
+Print the generated code
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If you’d like to run the same code multiple times, then it can be
+a bit tedious to step to the right code with ``pdb``. In that case, one
+approach is to simply copy-paste the generated ``forward`` pass into
+your code and examine it from there.
+
+::
+
+    # Module whose trace we want to debug
+    class M(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.x = torch.nn.Parameter(torch.rand(4, 3))
+
+        def forward(self, y):
+            return self.x + y
+
+    # Symbolically trace an instance of `M`
+    traced = symbolic_trace(M())
+
+    # Print the code generated from symbolic tracing. This outputs:
+    #   def forward(self, y):
+    #       x = self.x
+    #       add_1 = x + y;  x = y = None
+    #       return add_1
+    # Copy this code for later
```

Comment:
To show program outputs I prefer to use triple quotes
```
"""
   def forward(self, y):
       x = self.x
       add_1 = x + y;  x = y = None
       return add_1
"""
```

Or a separate code block in the RST